trieste/acquisition/function.py

```diff
@@ -124,10 +124,13 @@ def expected_improvement(model: ModelInterface, eta: tf.Tensor, at: QueryPoints)
        }
 
     :param model: The model of the objective function.
-    :param eta: The "best" observation.
+    :param eta: The "best" observation, with shape [1].
     :param at: The points for which to calculate the expected improvement.
     :return: The expected improvement at ``at``.
+    :raise ValueError (or InvalidArgumentError): If eta has the wrong shape.
     """
+    tf.debugging.assert_shapes([(eta, [1])], "eta has an invalid shape")
+
```

Comment:
can we instead check that eta is either shape [1] or `at.shape[-1]`? I actually have a case with Manchester for that

Reply:
this shape could actually be ['_', 1], but i thought i'd just keep things unambiguous